src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/PosixSignalRegistration.Windows.cs

```diff
@@ -49,17 +51,25 @@ private unsafe void Unregister()
                 {
                     _token = null;
 
-                    s_registrations.Remove(token);
-                    if (s_registrations.Count == 0 &&
-                        !Interop.Kernel32.SetConsoleCtrlHandler(&HandlerRoutine, Add: false))
+                    if (s_registrations.TryGetValue(token.SigNo, out List<Token>? tokens))
                     {
-                        // Ignore errors due to the handler no longer being registered; this can happen, for example, with
-                        // direct use of Alloc/Attach/FreeConsole which result in the table of control handlers being reset.
-                        // Throw for everything else.
-                        int error = Marshal.GetLastPInvokeError();
-                        if (error != Interop.Errors.ERROR_INVALID_PARAMETER)
+                        tokens.Remove(token);
```

Comment:
If I understand this correctly, `token` will almost always be at the end of this list. Do we want to search from the end to avoid quadratic behavior when removing all the tokens? It probably won't matter since there are likely < 10 signal handlers registered in the average application.

Reply:
Ah, nvm, I see no reason to assume that an arbitrary call to Unregister should be trying to remove the most recently registered signal.